File: examples/go/guides/parallelization/mock_llm.go
package main

import "strings"

func MockGenerateContent(message string) string {
	return "Here is a helpful response to: " + message
}

type SafetyResult struct {
	Safe   bool
	Reason string
}

func MockSafetyCheck(message string) SafetyResult {
	if strings.Contains(strings.ToLower(message), "unsafe") {
		return SafetyResult{Safe: false, Reason: "Content flagged as potentially unsafe."}
	}
	return SafetyResult{Safe: true, Reason: "Content is appropriate."}
}

type EvalResult struct {
	Score    float64
	Approved bool
}

func MockEvaluateContent(content string) EvalResult {
	score := 0.3
	if len(content) > 20 {
		score = 0.85
	}
	return EvalResult{Score: score, Approved: score >= 0.7}
}

File: examples/ruby/guides/parallelization/mock_llm.rb
# frozen_string_literal: true

def mock_generate_content(message)
  "Here is a helpful response to: #{message}"
end

def mock_safety_check(message)
  if message.downcase.include?('unsafe')
    { 'safe' => false, 'reason' => 'Content flagged as potentially unsafe.' }
  else
    { 'safe' => true, 'reason' => 'Content is appropriate.' }
  end
end

def mock_evaluate_content(content)
  score = content.length > 20 ? 0.85 : 0.3
  { 'score' => score, 'approved' => score >= 0.7 }
end
